### PR TITLE
Improve performance for portable (again)

### PR DESCRIPTION
**Pull request by GabrielRavier**
From my measurements, this reduces the CPU usage of MakePixelWaveData by 10% (from 21% to 19% CPU usage).

### CODE REVIEW

**Comment by Clownacy** [comment]
I really don't think this is necessary. This is a function that's only ran during startup, and I'd rather keep compiler-specific hacks and edits to the base game code to a minimum. Maybe I should write-up a `PHILOSOPHY.md` file or something.

**Comment by GabrielRavier** [comment]
I mean, it does have a significant impact, at least on my machine, considering when debugging with things like valgrind it takes literal minutes for the game to start up (incidentally, it does still take more than a second for me to get to the intro in general)

**Comment by GabrielRavier** [comment]
I did more precise measurements, and I get an average startup time of 1.231 seconds of bootup time with this PR, and 1.398 without.

**Comment by GabrielRavier** [comment]
I've added a few things :
- OpenMP support is activated when available (will not break anything on non-OpenMP supporting platforms, and OpenMP pragmas are standardly ignored on other platforms anyway)
- Moved attributes to Attributes.h and added a few new ones. You can remove the ones that are unused if you really want, but it would only be annoying for me to recreate them later if i was to need them (I added them because I was trying to improve performance through their use)
- Marked Backend_Blit and Backend_ColourFill in the Software backend as hot, since ihey take 30% of processing time together
- Added the "UNLIKELY" attribute to a test within Backend_Blit. This improves performance of that function by 20% and the global performance by 3%
- Added the "ATTRIBUTE_OPTIMIZE("-ffast-math")" attribute to MakePixelWaveData. This could theoretically break things if the function depended on strict IEEE 754, but I have observed no differences from applying this (apart from a global 1% performance boost)

**Comment by Clownacy** [comment]
I'm still not sold on improving startup times. Outside of absurdly weak hardware, or weird debugging setups, it feels like micro-optimisation.

**Comment by GabrielRavier** [comment]
I can remove that part. Incidentally, I'm just going to remove the `ATTRIBUTE_OPTIMIZE` thing too since it only works on gcc (clang ignores it), gcc considers it deprecated and gives only a barely measurable ~1% of performance difference (took 10 minutes to be able to be sure it actually changed anything)

File: CMakeLists.txt [hunk 1]
```diff
@@ -477,6 +477,13 @@ if(NOT FORCE_LOCAL_LIBS)
 	find_package(SDL2)
 endif()
 
+# Enable OpenMP if available
+find_package(OpenMP)
+if (OpenMP_CXX_FOUND)
+	target_link_libraries(CSE2 OpenMP::OpenMP_CXX)
+	target_compile_definitions(CSE2 PRIVATE USE_OPENMP)
+endif()
+
```
Comment (Clownacy) [inline CMakeLists.txt]: What is this for?

Comment (GabrielRavier) [reply CMakeLists.txt]: Oh, nothing really at this point. I was just trying to do stuff with OpenMP, but I didn't manage to get it working. Removing this...

**Comment by GabrielRavier** [comment]
Oh and I solved the conflict too

**Comment by GabrielRavier** [comment]
Btw now it compiles and works properly (as far as I can see)

**Comment by GabrielRavier** [comment]
Solved the merge conflict

**Comment by GabrielRavier** [comment]
Solved the merge conflict again

**Comment by GabrielRavier** [comment]
I updated the change list btw

**Pull request merged**